cdap-runtime-ext-dataproc/src/main/java/io/cdap/cdap/runtime/spi/provisioner/dataproc/ComputeEngineCredentials.java

```diff
@@ -187,9 +159,35 @@ private AccessToken fetchAccessToken(String endPoint) throws IOException {
 
   @Override
   public AccessToken refreshAccessToken() throws IOException {
-    if (endPoint != null) {
-      return refreshAccessTokenRemotely(endPoint);
+    ExponentialBackOff backOff = new ExponentialBackOff.Builder()
+      .setInitialIntervalMillis(MIN_WAIT_TIME_MILLISECOND)
+      .setMaxIntervalMillis(MAX_WAIT_TIME_MILLISECOND).build();
+
+    Exception exception = null;
+    int counter = 0;
+    while (counter < NUMBER_OF_RETRIES) {
+      counter++;
+
+      try {
+        if (endPoint != null) {
+          return getAccessTokenRemotely(endPoint);
+        }
+        return getAccessTokenLocally();
+
+      } catch (Exception ex) {
+        // exception does not get logged since it might get too chatty.
+        exception = ex;
+      }
+
+      try {
+        Thread.sleep(backOff.nextBackOffMillis());
+      } catch (InterruptedException ex) {
+        exception = ex;
+        break;
+      }
     }
-    return refreshAccessTokenLocally();
+
+    LOG.error("Failed to fetch GoogleCredentials after {} retries", counter, exception);
```

Comment:
Don't log and throw exception. Logging should be done at the outermost level without rethrowing.